cmake/modules/RootConfiguration.cmake

```diff
@@ -620,7 +620,9 @@ get_filename_component(altcxx ${CMAKE_CXX_COMPILER} NAME)
 get_filename_component(altf77 "${CMAKE_Fortran_COMPILER}" NAME)
 get_filename_component(altld ${CMAKE_CXX_COMPILER} NAME)
 
-set(pythonvers ${PYTHON_VERSION_STRING})
+set(pythonvers ${PYTHON_VERSION_STRING_Development_Main})
```

Comment:
Here the decision is made that `--python-version` will print the highest version PyROOT has been built with, if any. But now it can happen that ROOT and PyROOT are built with different Python versions (ROOT only needs the Python interpreter, while PyROOT needs development too).

An alternative would be to keep `--python-version` for the ROOT Python version, and add `--pyroot-version2` and `--pyroot-version3` (note the `pyroot` and not `python` part) for the versions of PyROOT. Opinions @oshadura @Axel-Naumann @amadio ?

Reply:
I just used LCG dev3 on lxplus, and simply it prints python3 `--python-version` while pyroot was build with 2.7. It is confusing, and as you saw from post at ROOT forum I cc-ed you, absolutely not useful.

Reply:
other problem is with a new naming policy e.g `cpyy_$..UNDER_VERSION` is impossible to find with which python version was build pyroot (3_6?3_7?).

Reply:
> I just used LCG dev3 on lxplus, and simply it prints python3 --python-version while pyroot was build with 2.7. It is confusing, and as you saw from post at ROOT forum I cc-ed you, absolutely not useful.

So I understand you want `--python-version` to print the highest PyROOT version, let's hear the others.



> other problem is with a new naming policy e.g cpyy_$..UNDER_VERSION is impossible to find with which python version was build pyroot (3_6?3_7?).

Why is it impossible? We have all the variables required to find the Python versions that PyROOT was built with, you are doing it here in this code.

Reply:
@etejedor what would be the use case for users wanting to see "the ROOT Python version" when invoking `--python-version`?

I find it ambiguous, unless there is only one Python version that ROOT is building against, in which case `--python-version` could return that version. Else error? But waiting for @amadio to share the use case he has in mind for `--python-version`.

Reply:
@Axel-Naumann I don't think users would be particularly interested in knowing the Python version ROOT was built with, I agree. I just wanted to make sure we agree that `--python-version` will always refer to PyROOT, and on what we do if PyROOT was not built. In my opinion, if PyROOT is not built, `--python-version` should not return any version even if ROOT had one (to be consistent with its semantics if there is PyROOT).

Reply:
https://sft.its.cern.ch/jira/browse/ROOT-10293

Reply:
@Axel-Naumann output now: ```  oksana@oksana-dellxps  ~/CERN_sources/root/builds   fix-python-output ●  root-config --python-version
3.8.3
 oksana@oksana-dellxps  ~/CERN_sources/root/builds   fix-python-output ●  root-config --python2-version
2.7.18
 oksana@oksana-dellxps  ~/CERN_sources/root/builds   fix-python-output ●  root-config --python3-version
3.8.3
```